### PR TITLE
Comment Functionality

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
 - Have added commenting functionality
 - User can comment and then see a list of comments for each post

### CODE REVIEW

File: FoodFriends/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -59,7 +59,8 @@
         </activity>
         <activity
```
Comment (nammyfb) [inline FoodFriends/app/src/main/AndroidManifest.xml:60]: Is the app gonna be able to handle landscape view? Might be good to force portrait mode :)

File: FoodFriends/app/src/main/java/com/example/foodfriends/adapters/CommentsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        /**
+         * Binds the xml elements to comment data
+         * */
+        public void bind(Comment comment) {
+            //bind data to view elements
+            tvUsername.setText(comment.getUser().getUsername());
+            tvCommentBody.setText(comment.getText());
+            RequestOptions requestOptions = new RequestOptions();
+            requestOptions = requestOptions.transforms(new CenterCrop(), new RoundedCorners(65));
+            ParseFile pfp_url = new UserObservable(comment.getUser()).getProfilePhoto();
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/adapters/CommentsAdapter.java:81]: Camel case :)

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java [hunk 3]
```diff
@@ -75,15 +93,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
 
         tvPrice.setText("PRICE: " + restaurant.getPrice());
         tvRName.setText(restaurant.getName());
         tvAddress.setText(restaurant.getAddress());
         tvLikeCount.setText(String.valueOf(restaurant.getLikes()));
         tvToGoCount.setText(String.valueOf(restaurant.getTogos()));
-        RequestOptions requestOptions = new RequestOptions();
-        requestOptions = requestOptions.transforms(new CenterCrop(), new RoundedCorners(15));
-        Glide.with(getContext()).applyDefaultRequestOptions(requestOptions).load(restaurant.getImageUrl()).into(ivRPic);
+        RequestOptions rPicOptions = new RequestOptions();
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:105]: Unnecessary  to prefix, with r :p

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java [hunk 4]
```diff
@@ -126,10 +157,35 @@ public void onClick(View v) {
...
                 helper.goToGmaps();
                 break;
+            case R.id.ivCommentSubmit:
+                add_comment();
         }
 
     }
 
+    private void add_comment() {
+        Comment new_comment = new Comment();
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:167]: too pythony :p

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java [hunk 5]
```diff
@@ -75,15 +93,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
         tvRName.setText(restaurant.getName());
         tvAddress.setText(restaurant.getAddress());
         tvLikeCount.setText(String.valueOf(restaurant.getLikes()));
         tvToGoCount.setText(String.valueOf(restaurant.getTogos()));
-        RequestOptions requestOptions = new RequestOptions();
-        requestOptions = requestOptions.transforms(new CenterCrop(), new RoundedCorners(15));
-        Glide.with(getContext()).applyDefaultRequestOptions(requestOptions).load(restaurant.getImageUrl()).into(ivRPic);
+        RequestOptions rPicOptions = new RequestOptions();
+        rPicOptions = rPicOptions.transforms(new CenterCrop(), new RoundedCorners(15));
+        Glide.with(getContext()).applyDefaultRequestOptions(rPicOptions).load(restaurant.getImageUrl()).into(ivRPic);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:107]: This block seems pretty important? Would suggest creating a function

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java [hunk 6]
```diff
@@ -75,15 +93,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         ivLike = view.findViewById(R.id.ivDetailLike);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:93]: General practice is views should be binded in the onCreate method.

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java [hunk 7]
```diff
@@ -158,4 +214,39 @@ public void update(Observable o, Object arg) {
...
     }
+
+    /**
+     * Adds all the restaurant's comments to the list
+     * */
+    private void queryComments() {
+        // specify what type of data we want to query - Post.class
+        ParseQuery<Comment> query = ParseQuery.getQuery(Comment.class);
+
+        query.whereEqualTo("restaurant", restaurant.getRestaurant());
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:225]: There could be a class that controls the commenting querying? Like CommentFetchHelper or some sort and would have methods that are responsible for saving/fetching comments?  

`public function queryForComments(restaurant, Callback/Listener for the return comments)`

**Review by nammyfb — CHANGES_REQUESTED** [review]

File: FoodFriends/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/AndroidManifest.xml:60]: Set portrait for every screen except for Map Activity!

File: FoodFriends/app/src/main/java/com/example/foodfriends/adapters/CommentsAdapter.java (hunk 2 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/adapters/CommentsAdapter.java:81]: Done!

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java (hunk 4 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/fragments/RestaurantDetailFragment.java:167]: sorry!

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/CommentListener.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.example.foodfriends.misc;
+
+public class CommentListener implements Listener {
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/misc/CommentListener.java:3]: Minor: This could be a class inside the CommentsFetchHelper

**Review by nammyfb — APPROVED** [review]

**Pull request merged**